src/file-manager/fm-directory-view.c

```diff
@@ -6421,6 +6421,87 @@ real_action_redo (FMDirectoryView *view)
 	caja_undostack_manager_redo (manager, GTK_WIDGET (view), finish_undoredo_callback);
 }
 
+static const gchar *known_bulk_renamers[][2] = {
+    { "bulky",  "bulky"     },
+    { "thunar", "thunar -B" }
+};
+
+static char *
+get_bulk_rename_tool (void)
+{
+    gchar *bulk_rename_tool = NULL;
+    gint i;
+
+    for (i = 0; i < G_N_ELEMENTS (known_bulk_renamers); i++) {
+        gchar *binary = g_find_program_in_path (known_bulk_renamers[i][0]);
+        if (binary != NULL) {
+            bulk_rename_tool = g_strdup (known_bulk_renamers[i][1]);
+            g_free (binary);
+            break;
+        }
+    }
+
+    if (bulk_rename_tool == NULL) {
+        g_settings_get (caja_preferences,
+                        CAJA_PREFERENCES_BULK_RENAME_TOOL,
+                        "^ay", &bulk_rename_tool);
+    }
+
+    return g_strstrip (bulk_rename_tool);
+}
+
+static gboolean
+have_bulk_rename_tool (void)
+{
+    gchar *bulk_rename_tool;
+    gboolean have_tool;
+
+    bulk_rename_tool = get_bulk_rename_tool ();
+    have_tool = ((bulk_rename_tool != NULL) && (*bulk_rename_tool != '\0'));
+    g_free (bulk_rename_tool);
+    return have_tool;
+}
+
+static void
+invoke_external_bulk_rename_utility (FMDirectoryView *view,
+                                     GList           *selection)
+{
+    GString *cmd;
+    char *parameter;
+    char *quoted_parameter;
+    char *bulk_rename_tool;
+    GList *walk;
+    CajaFile *file;
+
+    /* assemble command line */
+    bulk_rename_tool = get_bulk_rename_tool ();
+    cmd = g_string_new (bulk_rename_tool);
+    g_free (bulk_rename_tool);
+    for (walk = selection; walk; walk = walk->next) {
+        file = walk->data;
+        parameter = caja_file_get_uri (file);
+        quoted_parameter = g_shell_quote (parameter);
+        g_free (parameter);
+        cmd = g_string_append (cmd, " ");
+        cmd = g_string_append (cmd, quoted_parameter);
+        g_free (quoted_parameter);
+    }
+
+        guint i = 0;
```

Comment:
this `i` can be in the for-loop

Reply:
I cleaned up the legacy code a bit (I should in nemo also), though I'm not really keen on declaring i in the loop init - it's not consistent with the rest of caja (and personally I think it worsens readability, but I'm old and blind).